src/main/java/core/basesyntax/service/operations/PurchaseOperation.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.service.operations;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import java.util.List;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void getOperation(List<FruitTransaction> transactions) {
+
+        // someone has bought some fruits
+        List<FruitTransaction> transactionsSelected = transactions.stream()
+                .filter(a -> a.getOperation() == FruitTransaction.Operation.PURCHASE)
+                .toList();
+
+        for (FruitTransaction fruitTransaction : transactionsSelected) {
+            int newValue = Storage.fruits.get(fruitTransaction.getFruit())
+                    - fruitTransaction.getQuantity();
+            Storage.fruits.put(fruitTransaction.getFruit(), newValue);

```

Comment:
Before updating the storage, you need to check if the resulting balance after the purchase is positive. If the balance is negative, throw a RuntimeException, as specified in the checklist.